File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int l = 0;
        HashSet <char> hash = new  HashSet<char>();
        int res =0;
        for(int r=0; r<s.Length;r++){
                while (hash.Contains(s[r]))
                {
                    hash.Remove(s[l]);
                    l++;
                }

                hash.Add(s[r]);
                res = Math.Max(res,r-l+1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l = 0
        r = 0
        count_dict = {}
        maxf = 0
        res = 0

        while (r<len(s)):
            count_dict [s[r]] = 1 + count_dict.get(s[r],0)
            maxf = max(count_dict[s[r]],maxf)
            while r-l+1-maxf > k:
                count_dict[s[l]] -= 1
                l += 1
            res = max(r-l+1,res)
            r += 1
        return res





        